app/flatpak-builtins-utils.c

```diff
@@ -1065,6 +1077,23 @@ as_store_find_app (AsMetadata *mdata,
   for (j = 0; j < 2; j++)
     {
       const char *id = j == 0 ? appid : desktopid;
+#if AS_CHECK_VERSION(1, 0, 0)
+      AsComponentBox *components = as_metadata_get_components (mdata);
+
+      for (gsize i = 0; i < as_component_box_len (components); i++)
+        {
+          AsComponent *cpt = as_component_box_index (components, i);
+          AsBundle *bundle;
+
+          if (g_strcmp0 (as_component_get_id (cpt), id) != 0)
+            continue;
+
+          bundle = as_component_get_bundle (cpt, AS_BUNDLE_KIND_FLATPAK);
+          if (bundle &&
+              g_str_equal (as_bundle_get_id (bundle), ref))
+            return cpt;
```

Comment:
```suggestion
          AsComponent *component = as_component_box_index (components, i);
          AsBundle *bundle;

          if (g_strcmp0 (as_component_get_id (component), id) != 0)
            continue;

          bundle = as_component_get_bundle (component, AS_BUNDLE_KIND_FLATPAK);
          if (bundle &&
              g_str_equal (as_bundle_get_id (bundle), ref))
            return component;
```

Minor nit, I don't think the shorthand is necessary.

Reply:
This bit is also inconsistent with other areas calling it `app` but this seems more accurate anyway.

Reply:
I just called it what it was called previously :-) Since at no point this code ever seems to filter the components for applications-only, it would actually overall be more accurate to rename it to component/cpt everywhere.
I just kept it close to the original code for now (but could definitely rename it if that's wanted!).

Reply:
Yeah, `component` makes sense to me.

Reply:
In AppStream if it's a variable we just call it `cpt` to save on typing - in function names or anything that's API, it's always spelled out in full.

I'll be back on monday and will make the adjustments then :-)